app/models/checkbox_criterion.rb

```diff
@@ -57,7 +48,7 @@ def self.create_or_update_from_csv_row(row, assignment)
 
     # If a CheckboxCriterion with the same name exists, load it up. Otherwise,
     # create a new one.
-    criterion = assignment.get_criteria(:all, :checkbox).find_or_create_by(name: name)
+    criterion = assignment.criteria.where(type: 'CheckboxCriterion').find_or_create_by(name: name)
```

Comment:
Ah, simplify this by removing the `where` and doing `find_or_create_by(name: ..., type: 'CheckboxCriterion')`